src/lang/KclSingleton.ts

```diff
@@ -232,6 +233,12 @@ export class KclManager {
   async executeAst(args: ExecuteArgs = {}): Promise<void> {
     if (this.isExecuting) {
       this.executeIsStale = args
+      // Previous execution will be rejected and finish, execution will be marked as stale
+      // new executeAst will start. This will reject previous AST calls while the user is typing
```

Comment:
Definitely a nit, because I'm talking about a comment, but "AST calls" doesn't seem like the right terminology. Maybe I misunderstood but would "will reject pending engine commands" make sense?

Reply:
The reason I mention `AST` at all is the code that ends up returning is this block

```
    const { logs, errors, programMemory } = await executeAst({
      ast: newAst,
      engineCommandManager: this.engineCommandManager,
      useFakeExecutor: true,
    })
```

You are correct that the pending engine commands will be rejected but higher up the only reason those commands are being created is the `executeAst` call.

I can update the comment.

Reply:
hmm okay, no you can leave it. I just don't know what an "AST call" is, would "AST execution" work?